server/util/db/db.go

```diff
@@ -1135,3 +1140,29 @@ func ScanRows[T any](rq interfaces.DBRawQuery, fn func(ctx context.Context, val
 		return nil
 	})
 }
+
+// ScanAll executes the given query and scans all the resulting rows into a
+// slice of the specified type.
+//
+// This should not be used if you expect the result set size to be large.
+//
+// Example:
+//
+//	foos, err := db.All(rq, &tables.Foo{})
+//
+// TODO(vadim): check if there are any ScanRows calls that can be converted to ScanAll.
+func ScanAll[T any](rq interfaces.DBRawQuery, t *T) ([]*T, error) {
```

Comment:
imo `ScanRows` vs. `ScanAll` is a bit unclear - maybe something like `CollectRows` or `ScanToSlice` could make sense?

Reply:
I renamed ScanRows to ScanEach. Hope that differentiates it enough.